airflow/providers_manager.py

```diff
@@ -467,7 +468,7 @@ def _init_airflow_core_hooks(self):
                 connection_type=None,
                 connection_testable=False,
             )
-        for cls in [FSHook, PackageIndexHook]:
+        for cls in [FSHook, PackageIndexHook, PythonVirtualenvHook]:
```

Comment:
Actually it would be the right place to add. Have not seen this error before.

Reply:
@jscheffl
I found the reason!
The validators include InputRequired for the field:
```python
"requirements": StringField(  
    lazy_gettext("Requirements"),  
    description=lazy_gettext("Python Requirements follows PEP 508 format."),  
    widget=BS3TextAreaFieldWidget(),  
    validators=[InputRequired()],    # <- here
)
```

Because of this, when a POST request is made to `/connection/add` by flask_appbuilder, the required field(`extra__python_venv__requirements`) is not filled and so shows an alert, not "Added Row"(success message)


> here is the html response in https://github.com/apache/airflow/blob/a6c88302bdb46702f00568db4fdc709c31bb69f6/tests/www/views/test_views_connection.py#L55-L59 

```html
...
            <textarea class="form-control" id="extra__python_venv__requirements" name="extra__python_venv__requirements" placeholder="Requirements" required rows="3">
</textarea>
              <span class="help-block">Python Requirements follows PEP 508 format.</span>
          
            <div class="alert alert-danger">
            
              This field is required.
            
            </div>
...
```